src/page-multi-facility/FacilityInputForm.tsx

```diff
@@ -136,21 +130,28 @@ const PageHeaderContainer = styled.div`
 
 interface Props {
   scenarioId: string;
+  facility: Facility | undefined;
+  rtData: RtDataMapping;
 }
 
-const FacilityInputForm: React.FC<Props> = ({ scenarioId }) => {
+const FacilityInputForm: React.FC<Props> = ({
+  rtData,
+  facility,
```

Comment:
I originally had it use the context variables in `FacilityInputForm`, but ran into some issues with duplicating the facility. The `duplicateFacility` function dispatches the `selectFacility` action which will update the context variable `selectedFacilityId`, but since we are setting `facilityName`, `description` and `systemType` as internal state for the inputs, these will not update to the new facility's values because the component doesn't mount/unmount when context changes. This is also the reason I am passing a `key` prop to the `FacilityInputForm` from the parent `FacilityPage`, to force it to mount/unmount when `selectedFacilityId` changes. 

Let me know if you have any ideas around that...another option I think might work is to have a `useEffect` that updates the state variables when the context changes.

Reply:
interesting. setting a `key` to force a remount when the facility changes isn't enough to fix that on its own? I would think that would cause all of the state variables to be reset.

Reply:
Oh yeah, I could set the key in the parent and then access the facility again from the context. I'll try that!